pkg/util/workqueue/queue.go

```diff
@@ -113,6 +113,17 @@ func (q *Type) Get() (item interface{}, shutdown bool) {
 	return item, false
 }
 
+// Get the head of the queue
+func (q *Type) Peek() interface{} {
```

Comment:
Peek is not a good name, if it's returning head, then name it `Head`. Additionally you're missing a test-case for this method.